### DISCUSSION TITLE
Conditional fits with fixed non-POI parameters

### DISCUSSION THREAD

**Discussion by balunas** [scikit-hep/pyhf#1603] [Q&A] [answered]
I've got a question which feels like it should very simple, but I can't seem to find any indication of how to do it in the documentation. I've got a likelihood model with a pretty common structure: some yields in bins from different sources, which each have a few modifiers attached. These represent systematic NPs or unconstrained normalization parameters in the fit, etc. I want to run a simple MLE fit holding some of these parameters fixed at values that I specify at the python level (and allowing all the others to vary as usual). I know that there's a dedicated function that does this holding the POI fixed, but I want to fix _other_ parameters. I'd be surprised if the python API doesn't support this, but I can't seem to find it documented anywhere. Does anyone know how to do this?

**Comment by alexander-held** [comment, marked as answer]
The [`infer.mle.fit`](https://pyhf.readthedocs.io/en/v0.6.3/_generated/pyhf.infer.mle.fit.html#pyhf.infer.mle.fit) API takes a list of initial parameter values and a list of booleans to denote whether a given parameter is meant to be held constant. If it is set to constant, it will take the initial parameter value. So you can use [`model.config.par_names`](https://pyhf.readthedocs.io/en/v0.6.3/_generated/pyhf.pdf._ModelConfig.html#pyhf.pdf._ModelConfig.par_names) to identify the position in the list of parameters that a given parameter you want to hold constant is located at, then use [`model.config.suggested_init()`](https://pyhf.readthedocs.io/en/v0.6.3/_generated/pyhf.pdf._ModelConfig.html#pyhf.pdf._ModelConfig.suggested_init) and [`model.config.suggested_fixed()`](https://pyhf.readthedocs.io/en/v0.6.3/_generated/pyhf.pdf._ModelConfig.html#pyhf.pdf._ModelConfig.suggested_fixed) to get the default settings for initial values / fixed parameters, edit these lists as desired (parameter order matches `model.config.par_names`), and pass the new lists to `infer.mle.fit` while using the `init_pars` and `fixed_params` keyword arguments.

**Reply by balunas** [reply]
Perfect, thanks - it was there in the API all along, I apparently just missed it or was looking in the wrong place. Sorry for the silly question!
